### PR TITLE
feat: multi-activity revert + first-run flow API

### PR DESCRIPTION
**Pull request by Lemkinator**
## Summary
- Reverts single-activity + Navigation Component migration; restores activity-based screens (OOBE, About, AboutMe, Settings, Libs)
- Adds `FirstRunFlowUtils.kt`: `handleFirstRun` / `advanceFirstRun` / `isFirstRunStep` / `setupFirstRunFlow` — a reusable, ordered chain of task-root activities (OOBE-first) replacing the deprecated `checkAppStartAndHandleOOBE` / `openOOBEAndFinish` helpers
- Wires `CommonUtilsOOBEActivity` to advance the chain; removes fragment/nav code and deprecated OOBE companion state

## Test Plan
- [ ] `./gradlew :lib:build` → BUILD SUCCESSFUL (spotless, detekt, lint, unit tests)
- [ ] `FirstRunFlowTest`: 4 tests pass (chain ordering, unregistered-activity guard)

## Notes
- **Merge this before GetIcon PR** — GetIcon consumes the new API via `includeBuild("../common-utils")`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Multi-Activity Revert with Stateless First-Run / Onboarding Flow

This PR reverts the single-activity + Navigation Component migration and restores activity-based screens (OOBE, About, AboutMe, Settings, Libs). It introduces a stateless, ordered first-run/onboarding chain with a small API surface implemented in OnboardingUtils.kt:

- setupOnboarding(steps) — register post‑OOBE activity steps (OOBE remains implicit)
- onboardIfNeeded(versionCode, versionName, allowSkip) — launcher call; returns null when launching OOBE
- advanceOnboarding() — advance via intent-extras listing steps; commits accepted TOS only after the final step and starts main activity
- isOnboardingStep() — detect chain-launched step activities

Notable changes
- Removed navigation graph (commonutils_navigation.xml) and nav-fragment helpers; restored activity classes and removed deprecated `@Deprecated` annotations.
- Removed deprecated OOBE companion-state API and helpers (setAcceptedTosVersion, nextActivity, onContinue, checkAppStartAndHandleOOBE, openOOBEAndFinish).
- checkAppStart signature simplified (removed versionCodeThreshold).
- CommonUtilsOOBEActivity proceed flow now uses advanceOnboarding(); companion API reduced to tosChanged.
- Added Activity transition helpers: overrideFadeOpenTransition() and finishWithFade() (pre-/post-API34 ordering fixed).
- Hardened onboarding guards:
  - isFirstTimeVersion / versionThresholdPassed return false on fresh installs (lastVersionCode == -1).
  - advanceOnboarding now bails (logs + finishWithFade) when current activity is not found in the chain instead of silently completing.
  - Onboarding.steps setter made internal to force mutations via setupOnboarding().
- Tests:
  - OnboardingUtilsTest verifying chain ordering and unregistered-activity guard.
  - AppStartTest adjusted/renamed and adds first-install edge-case assertions.
- Docs: CLAUDE.md / docs updated describing first-run behavior and recommended call-site pattern.

Cleanup
- Removed several file-level `@Suppress`("unused") annotations and minor formatting/lint/test tweaks.

Test plan
- Run ./gradlew :lib:build (spotless, detekt, lint, unit tests) and the onboarding/first-run tests.

Notes
- Onboarding is stateless: step advancement is driven by intent extras (ordered class-name list); final commit happens only after the last step.
- Merge before GetIcon PR (GetIcon will consume this API via includeBuild("../common-utils")).

### CODE REVIEW

File: lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+): Boolean {
+    if (allowSkip && intent.getBooleanExtra(EXTRA_SKIP_FIRST_RUN, false)) return false
+    if (!checkAppStart(versionCode, versionName).shouldShowOOBE) return false
+    FirstRunFlow.mainActivity = this::class.java
+    startActivity(Intent(this, CommonUtilsOOBEActivity::class.java).putExtra(EXTRA_FIRST_RUN, true))
+    @Suppress("DEPRECATION")
+    if (SDK_INT < UPSIDE_DOWN_CAKE) overridePendingTransition(fade_in, fade_out)
+    finishAfterTransition()
+    return true
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — `handleFirstRun` now returns `AppStart?` (null = first run/caller must return; non-null = proceed). `checkAppStart` is called exactly once and the result is returned directly, so callers can check `isFirstTimeVersion` etc. without a second invocation.

File: lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            checkNotNull(FirstRunFlow.mainActivity) {
+                "advanceFirstRun: mainActivity not set — call handleFirstRun() from the launcher activity before the chain starts"
+            }
+        commonUtilsSettings.acceptedTosVersion = resources.getInteger(R.integer.commonutils_tos_version)
+        startActivity(Intent(this, main))
+    }
+    @Suppress("DEPRECATION")
+    if (SDK_INT < UPSIDE_DOWN_CAKE) overridePendingTransition(fade_in, fade_out)
+    finishAfterTransition()
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — The flow is now fully stateless. `handleFirstRun` passes `EXTRA_FIRST_RUN_MAIN_ACTIVITY` (launcher class name) and `EXTRA_FIRST_RUN_STEPS` (step class names) as Intent extras to OOBE. `advanceFirstRun` reads and forwards those extras at each step. Intent extras survive process death, so the chain resumes correctly even if the OS kills the process mid-flow.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -29,7 +29,7 @@ ksp = "2.3.9"
 ktlint = "1.7.1"
 lottie = "6.7.1"
 material3 = "1.4.0"
-mockk = "1.14.9"
+mockk = "1.14.11"
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply gradle/libs.versions.toml:32]: 🤖 **Claude Code** · The bot's version model is running backwards — `1.14.11` is *newer* than `1.14.9`, not older. This PR aligned with the version already merged to main via PR #125. No change needed.

File: lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import android.os.Build.VERSION_CODES.UPSIDE_DOWN_CAKE
+import androidx.appcompat.app.AppCompatActivity
+import de.lemke.commonutils.data.commonUtilsSettings
+import de.lemke.commonutils.ui.activity.CommonUtilsOOBEActivity
+
+/** Intent extra marking an activity launched as a step of the first-run chain. */
+const val EXTRA_FIRST_RUN = "commonUtilsFirstRunStep"
+
+/** Intent extra (honored only when the host opts in) that bypasses the first-run chain, for benchmarks. */
+const val EXTRA_SKIP_FIRST_RUN = "commonUtilsSkipFirstRun"
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — `EXTRA_FIRST_RUN_MAIN_ACTIVITY` and `EXTRA_FIRST_RUN_STEPS` added as part of the stateless-flow refactor.

File: lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    var steps: List<Class<out Activity>> = emptyList()
+
+    /** The activity to land on once the chain completes (set by [handleFirstRun]). */
+    var mainActivity: Class<out Activity>? = null
+}
+
+/** Declares the app-specific first-run steps that run after OOBE. Apps with OOBE only may skip this. */
+fun setupFirstRunFlow(steps: List<Class<out Activity>> = emptyList()) {
+    FirstRunFlow.steps = steps
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — KDoc updated: recommends calling `setupFirstRunFlow` in `Application.onCreate()` to ensure steps survive process death.

[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — `setupFirstRunFlow` now rejects `CommonUtilsOOBEActivity` in steps and requires all entries to be unique.

File: lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+@Suppress("ReturnCount")
+fun AppCompatActivity.handleFirstRun(
+    versionCode: Int,
+    versionName: String,
+    allowSkip: Boolean = false,
+): Boolean {
+    if (allowSkip && intent.getBooleanExtra(EXTRA_SKIP_FIRST_RUN, false)) return false
+    if (!checkAppStart(versionCode, versionName).shouldShowOOBE) return false
+    FirstRunFlow.mainActivity = this::class.java
+    startActivity(Intent(this, CommonUtilsOOBEActivity::class.java).putExtra(EXTRA_FIRST_RUN, true))
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/FirstRunFlowUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — Same fix as above: chain is now stateless via Intent extras.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+ * step (as task root, tagged [EXTRA_ONBOARDING_STEP]); or, past the last step, commits the
+ * completion flag and starts the main activity.
+ *
+ * Call from a step activity when the user finishes that step.
+ */
+fun Activity.advanceOnboarding() {
+    val mainActivityName =
+        checkNotNull(intent.getStringExtra(EXTRA_ONBOARDING_MAIN_ACTIVITY)) {
+            "advanceOnboarding: EXTRA_ONBOARDING_MAIN_ACTIVITY missing — was this activity started by onboardIfNeeded?"
+        }
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — added `isOnboardingStep()` guard at the top of `advanceOnboarding`: if the activity wasn't launched as part of the chain it simply finishes with the transition animation instead of crashing on the missing extra.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+                putExtra(EXTRA_ONBOARDING_MAIN_ACTIVITY, mainActivityName)
+                putStringArrayListExtra(EXTRA_ONBOARDING_STEPS, ArrayList(stepsNames))
+            },
+        )
+    } else {
+        @Suppress("UNCHECKED_CAST")
+        val mainClass = Class.forName(mainActivityName) as Class<out Activity>
+        commonUtilsSettings.acceptedTosVersion = resources.getInteger(R.integer.commonutils_tos_version)
+        startActivity(Intent(this, mainClass))
+    }
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — replaced `Class.forName` + `@Suppress("UNCHECKED_CAST")` cast with `Intent.setClassName(this, className)` throughout `advanceOnboarding`.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+/** The full ordered chain: OOBE first, then the configured steps. */
+private fun onboardingChain(): List<Class<out Activity>> = listOf(CommonUtilsOOBEActivity::class.java) + Onboarding.steps
+
+/** Returns the step after [current] in the chain, or `null` if [current] is the last step. */
+internal fun nextOnboardingStep(current: Class<*>): Class<out Activity>? {
+    val chain = onboardingChain()
+    val index = chain.indexOfFirst { it == current }
+    if (index == -1) return null
+    return chain.getOrNull(index + 1)
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Won't fix — `nextOnboardingStep` is an intentional `internal` test helper that exposes the chain-resolution logic as a standalone unit-testable function. The KDoc now notes this explicitly. Production navigation runs through `advanceOnboarding`.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+        startActivity(
+            Intent(this, CommonUtilsOOBEActivity::class.java).apply {
+                putExtra(EXTRA_ONBOARDING_STEP, true)
+                putExtra(EXTRA_ONBOARDING_MAIN_ACTIVITY, this@onboardIfNeeded::class.java.name)
+                putStringArrayListExtra(EXTRA_ONBOARDING_STEPS, ArrayList(Onboarding.steps.map { it.name }))
+            },
+        )
+        @Suppress("DEPRECATION")
+        if (SDK_INT < UPSIDE_DOWN_CAKE) overridePendingTransition(fade_in, fade_out)
+        finishAfterTransition()
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — extracted public `Activity.finishWithFade()` in `TransformationUtils.kt`: uses `overrideActivityTransition(OVERRIDE_TRANSITION_CLOSE, fade_in, fade_out)` on API 34+ and the deprecated `overridePendingTransition` on older versions. Applied to all 3 occurrences in `OnboardingUtils` (including `onboardIfNeeded`, which Gemini missed). Function is public so apps can reuse it in their own activities.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+ * step (as task root, tagged [EXTRA_ONBOARDING_STEP]); or, past the last step, commits the
+ * completion flag and starts the main activity.
+ *
+ * Call from a step activity when the user finishes that step.
+ */
+fun Activity.advanceOnboarding() {
+    if (!isOnboardingStep()) {
+        @Suppress("DEPRECATION")
+        if (SDK_INT < UPSIDE_DOWN_CAKE) overridePendingTransition(fade_in, fade_out)
+        finishAfterTransition()
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — same fix as above (see reply on line 183).

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+                putStringArrayListExtra(EXTRA_ONBOARDING_STEPS, ArrayList(stepsNames))
+            },
+        )
+    } else {
+        commonUtilsSettings.acceptedTosVersion = resources.getInteger(R.integer.commonutils_tos_version)
+        startActivity(Intent().setClassName(this, mainActivityName))
+    }
+    @Suppress("DEPRECATION")
+    if (SDK_INT < UPSIDE_DOWN_CAKE) overridePendingTransition(fade_in, fade_out)
+    finishAfterTransition()
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — same fix as above (see reply on line 183).

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+    val isFirstTimeVersion get() = lastVersionCode < versionCode
+
+    /** `true` if the user has accepted the current TOS version. */
+    val tosAccepted get() = acceptedTosVersion >= tosVersion
+
+    /** `true` if OOBE should be shown (first install or TOS not accepted). */
+    val shouldShowOOBE get() = isFirstTime || !tosAccepted
+
+    /** `true` if [threshold] falls within the range of version codes updated across on this launch. */
+    fun versionThresholdPassed(threshold: Int) = threshold in lastVersionCode..<versionCode
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — added `lastVersionCode >= 0` guard (`in 0..<versionCode` range form) to `isFirstTimeVersion` and `lastVersionCode >= 0 &&` prefix to `versionThresholdPassed`. Both helpers now return `false` on fresh install (`lastVersionCode == -1`). Tests added for both cases.

File: lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt [hunk 14]
```diff
@@ -189,3 +193,14 @@ fun View.transformTo(
...
+
+/** Applies a fade-out close transition and finishes the activity (predictive-back aware). */
+fun Activity.finishWithFade() {
+    if (SDK_INT >= UPSIDE_DOWN_CAKE) {
+        overrideActivityTransition(Activity.OVERRIDE_TRANSITION_CLOSE, fade_in, fade_out)
+    } else {
+        @Suppress("DEPRECATION")
+        overridePendingTransition(fade_in, fade_out)
+    }
+    finishAfterTransition()
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt]: 🤖 **Claude Code** · Partially fixed ✓ — pre-API 34: swapped order so `finishAfterTransition()` is called before `overridePendingTransition()`, matching the documented post-finish API contract. API 34+: kept `overrideActivityTransition(OVERRIDE_TRANSITION_CLOSE)` before `finishAfterTransition()` — the Android docs explicitly state 'you may call this in any of the lifecycle methods', so this is valid and preferred over removing it (removing would mean no close animation on API 34+).

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+
+/** Checks whether this is the first run, a version upgrade, or a normal start; persists version info for next launch. */
+private fun AppCompatActivity.checkAppStart(
+    versionCode: Int,
+    versionName: String,
+): AppStart {
+    val lastVersionCode = commonUtilsSettings.lastVersionCode
+    val lastVersionName = commonUtilsSettings.lastVersionName
+    commonUtilsSettings.lastVersionCode = versionCode
+    commonUtilsSettings.lastVersionName = versionName
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — `checkAppStart` no longer eagerly commits `lastVersionCode`/`lastVersionName`. Version info is now committed in two places: `onboardIfNeeded` (normal start, no onboarding needed) and `advanceOnboarding` (chain completion). This ensures a crashed/killed process re-detects first-run/upgrade state correctly on restart.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+                putExtra(EXTRA_ONBOARDING_MAIN_ACTIVITY, this@onboardIfNeeded::class.java.name)
+                putStringArrayListExtra(EXTRA_ONBOARDING_STEPS, ArrayList(Onboarding.steps.map { it.name }))
+            },
+        )
+        finishWithFade()
+        null
+    } else {
+        appStart
+    }
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt:145]: 🤖 **Claude Code** · Fixed ✓ — `onboardIfNeeded` now checks for `EXTRA_ONBOARDING_APP_START_RESULT` in the launch intent. When present (main activity re-launched after chain completion), it reconstructs the original `AppStart` from the intent extras so callers can reliably check `isFirstTime` / `isFirstTimeVersion`. On normal start, version info is committed here. On onboarding start, the pending version and original AppStart fields are passed as extras to OOBE.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+                putExtra(EXTRA_ONBOARDING_MAIN_ACTIVITY, mainActivityName)
+                putStringArrayListExtra(EXTRA_ONBOARDING_STEPS, ArrayList(stepsNames))
+            },
+        )
+    } else {
+        commonUtilsSettings.acceptedTosVersion = resources.getInteger(R.integer.commonutils_tos_version)
+        startActivity(Intent().setClassName(this, mainActivityName))
+    }
+    finishWithFade()
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt:185]: 🤖 **Claude Code** · Fixed ✓ — `advanceOnboarding` now propagates the five AppStart/version extras through each step. On chain completion: commits `lastVersionCode`/`lastVersionName` from the pending extras, sets `acceptedTosVersion`, then launches the main activity with `EXTRA_ONBOARDING_APP_START_RESULT`/`LAST_VERSION_CODE`/`LAST_VERSION_NAME` so `onboardIfNeeded` can reconstruct the original `AppStart`.

File: lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt [hunk 18]
```diff
@@ -189,3 +193,25 @@ fun View.transformTo(
...
+
+/** Applies a fade-in open transition for this activity. Call in `onCreate` before `super`. */
+fun Activity.overrideFadeOpenTransition() {
+    if (SDK_INT >= UPSIDE_DOWN_CAKE) {
+        overrideActivityTransition(Activity.OVERRIDE_TRANSITION_OPEN, fade_in, fade_out)
+    } else {
+        @Suppress("DEPRECATION")
+        overridePendingTransition(fade_in, fade_out)
+    }
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — removed the pre-API 34 `overridePendingTransition` branch from `overrideFadeOpenTransition()`. `overridePendingTransition` must be called by the SOURCE activity after `startActivity()`; calling it in a destination `onCreate` is a no-op. Function is now API 34+ only with a 'no-op below API 34' note in the KDoc.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+        commitAppStart(versionCode, versionName)
+        return AppStart(
+            ctx.appStartResult,
+            versionCode,
+            versionName,
+            ctx.lastVersionCode,
+            ctx.lastVersionName,
+            resources.getInteger(R.integer.commonutils_tos_version),
+            commonUtilsSettings.acceptedTosVersion,
+        ).also { Log.d(TAG, "onboardIfNeeded (post-onboarding): $it") }
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — added `intent.removeExtra(EXTRA_ONBOARDING_CONTEXT)` before consuming the carrier so config-change recreation (rotation etc.) doesn't re-enter the post-onboarding path.

File: lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt [hunk 20]
```diff
@@ -189,3 +195,22 @@ fun View.transformTo(
         }
     }
 }
+
+/** Applies a fade-in open transition for this activity. Call in `onCreate`, after `super.onCreate`. */
+fun Activity.overrideFadeOpenTransition() {
+    if (SDK_INT >= UPSIDE_DOWN_CAKE) {
+        overrideActivityTransition(OVERRIDE_TRANSITION_OPEN, fade_in, fade_out)
+    }
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/TransformationUtils.kt:219]: 🤖 **Claude Code** · Won't fix — `overridePendingTransition` called in the **destination** activity's `onCreate` is a no-op on pre-API 34: the open animation is already scheduled by the time `onCreate` runs. Pre-34, only the **source** activity can override the open transition (calling `overridePendingTransition` after `startActivity`). This was already understood, explicitly noted in commit `50b27f1`, and removed in `b3e8bd0` following a prior CodeRabbit finding. The no-op on <34 is the only correct behavior for a destination-side function.

File: lib/src/main/java/de/lemke/commonutils/AppStartUtils.kt [hunk 21]
```diff
@@ -131,10 +49,44 @@ class AppStart(
...
+            lastVersionCode == -1 -> {
+                AppStartResult.FIRST_TIME
+            }
+
+            lastVersionCode < versionCode -> {
+                AppStartResult.FIRST_TIME_VERSION
+            }
+
+            lastVersionCode > versionCode -> {
+                Log.w(TAG, "Current version code ($versionCode) is less then the one recognized on last startup ($lastVersionCode). ")
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/AppStartUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — typo corrected in `AppStartUtils.kt` log message.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+            // Post-onboarding: main activity re-launched after chain completed — reconstruct original AppStart.
+            intent.removeExtra(EXTRA_ONBOARDING_CONTEXT)
+            AppStart(
+                ctx.appStartResult,
+                versionCode,
+                versionName,
+                ctx.lastVersionCode,
+                ctx.lastVersionName,
+                resources.getInteger(R.integer.commonutils_tos_version),
+                commonUtilsSettings.acceptedTosVersion,
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt]: 🤖 **Claude Code** · Fixed ✓ — in the post-onboarding branch, `commitAppStart` always writes `acceptedTosVersion = tosVersion`, so the reconstructed `AppStart` now uses `tosVersion` for both fields directly. Reading the stale stored value before commit ran was the bug; the carrier itself proves onboarding completed and TOS was accepted.

File: lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt [hunk 23]
```diff
@@ -0,0 +1,193 @@
...
+            completeOnboarding(ctx)
+        }
+    }
+    finishWithFade()
+}
+
+/** Starts the main activity with the carrier so `onboardIfNeeded` can commit and reconstruct `AppStart`. */
+private fun Activity.completeOnboarding(ctx: OnboardingContext) {
+    startActivity(Intent().setClassName(this, ctx.mainActivityName).putOnboardingContext(ctx))
+}
```
[the remark this answers is not in the document]
Comment (Lemkinator) [reply lib/src/main/java/de/lemke/commonutils/OnboardingUtils.kt:190]: 🤖 **Claude Code** · Won't fix — the loss only occurs on TOS bumps (not fresh installs, where no meaningful deep-link context exists). Carrying the original `Intent` as a `Parcelable` in `OnboardingContext` is the natural approach, but risks hitting the 1 MB Binder limit when extras are large, and Intent flags from the original launch can conflict with the chain's own task-stack flags. Callers that care about deep-link restoration after a TOS bump can handle it in `onNewIntent` / `onResume` — the library has no safe, general way to replay an arbitrary Intent across process boundaries.
